internal/store/postgres/billing_plan_repository.go

```diff
@@ -273,3 +349,112 @@ func (r BillingPlanRepository) List(ctx context.Context, filter plan.Filter) ([]
 	}
 	return plans, nil
 }
+
+func (r BillingPlanRepository) ListWithProducts(ctx context.Context, filter plan.Filter) ([]plan.Plan, error) {
+	pln := goqu.T(TABLE_BILLING_PLANS).As("plan")
+	prd := goqu.T(TABLE_BILLING_PRODUCTS).As("product")
+	stmt := dialect.From(pln).
+		Join(
+			prd,
+			goqu.On(
+				goqu.L("CAST(plan.id AS text)").Eq(goqu.L("ANY(product.plan_ids)")),
+			),
+		).Select(
+		pln.Col("id").As("plan_id"),
+		pln.Col("name").As("plan_name"),
+		pln.Col("title").As("plan_title"),
+		pln.Col("description").As("plan_description"),
+		pln.Col("interval").As("plan_interval"),
+		pln.Col("on_start_credits").As("plan_on_start_credits"),
+		pln.Col("state").As("plan_state"),
+		pln.Col("trial_days").As("plan_trial_days"),
+		pln.Col("metadata").As("plan_metadata"),
+		pln.Col("created_at").As("plan_created_at"),
+		pln.Col("updated_at").As("plan_updated_at"),
+		prd.Col("deleted_at").As("plan_deleted_at"),
+		prd.Col("id").As("product_id"),
+		prd.Col("provider_id").As("product_provider_id"),
+		prd.Col("name").As("product_name"),
+		prd.Col("title").As("product_title"),
+		prd.Col("description").As("product_description"),
+		prd.Col("title").As("product_behavior"),
+		prd.Col("config").As("product_config"),
+		prd.Col("state").As("product_state"),
+		prd.Col("metadata").As("product_metadata"),
+		prd.Col("created_at").As("product_created_at"),
+		prd.Col("updated_at").As("product_updated_at"),
+		prd.Col("deleted_at").As("product_deleted_at"),
+	)
+
+	var ids []string
+	var names []string
+	if len(filter.IDs) > 0 {
```

Comment:
the filter is taking an array but we filter only on 0th index?

Reply:
We check the ids field to see if we need to filter by name or an id. I thought of making it work only with ids, but decided to keep it consistent with the `List` method in this repository. 
If it is safe to remove it from there also, then I think we could make both methods work with only ids.